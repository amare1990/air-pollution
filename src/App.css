.App {
  text-align: center;
  min-width: 375px;
}
