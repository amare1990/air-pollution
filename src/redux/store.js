import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './current/current';
import cityReducer from './city/city';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
    city: cityReducer,
  },
});
