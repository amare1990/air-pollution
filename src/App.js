import { Routes, Route } from 'react-router-dom';
import './App.css';
import CityWeather from './components/CityWeather';
import Homepage from './components/Homepage';
import NavBar from './components/NavBar';

function App() {
  return (
    <div className="App">

      <Routes>
        <Route path="/" element={<NavBar />} />
        <Route path="Details" element={<CityWeather />} />
        {/* <Route path="categ" element={<Homepage />} /> */}
      </Routes>
      <Homepage />
    </div>
  );
}

export default App;
