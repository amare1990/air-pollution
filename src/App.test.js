import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';
import store from './redux/store';

test('Testing if App component renders correctly', () => {
  const appSnapshot = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(appSnapshot).toMatchSnapshot();
});
