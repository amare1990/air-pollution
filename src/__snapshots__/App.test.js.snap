// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if App component renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <a
      className=""
      href="/https://www.openweathermap.org"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo-img"
        src="logo_weather.png"
      />
    </a>
    <div
      className="search-bar-wrapper"
    >
      <input
        className="search-input"
        id=""
        onChange={[Function]}
        placeholder="search City"
        type="search"
        value=""
      />
      <button
        className="btn search-btn"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </header>
</div>
`;
